src/main/java/org/mifos/connector/mpesa/camel/routes/SafaricomRoutesBuilder.java

```diff
@@ -0,0 +1,166 @@
+package org.mifos.connector.mpesa.camel.routes;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.camel.Exchange;
+import org.apache.camel.LoggingLevel;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.model.dataformat.JsonLibrary;
+import org.json.JSONObject;
+import org.mifos.connector.mpesa.auth.AccessTokenStore;
+import org.mifos.connector.mpesa.dto.BuyGoodsPaymentRequestDTO;
+import org.mifos.connector.mpesa.dto.TransactionStatusRequestDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import static org.mifos.connector.mpesa.camel.config.CamelProperties.*;
+import static org.mifos.connector.mpesa.safaricom.config.SafaricomProperties.MPESA_BUY_GOODS_TRANSACTION_TYPE;
+import static org.mifos.connector.mpesa.utility.SafaricomUtils.getPassword;
+
+
+@Component
+public class SafaricomRoutesBuilder extends RouteBuilder {
+
+    @Value("${mpesa.api.passKey}")
+    private String passKey;
+
+    @Value("${mpesa.api.lipana}")
+    private String buyGoodsBaseUrl;
+
+    @Value("${mpesa.api.transaction-status}")
+    private String transactionStatusBaseUrl;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private AccessTokenStore accessTokenStore;
+
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Override
+    public void configure() {
+
+        /*
+         * Use this endpoint for getting the mpesa transaction status
+         * The request parameter is same as the safaricom standards
+         */
+        from("rest:POST:/buygoods/transactionstatus")
+                .id("buy-goods-transaction-status")
+                .process(exchange -> {
+                    String body = exchange.getIn().getBody(String.class);
+                    TransactionStatusRequestDTO transactionStatusRequestDTO = objectMapper.readValue(
+                            body, TransactionStatusRequestDTO.class);
+
+                    exchange.setProperty(BUY_GOODS_REQUEST_BODY, transactionStatusRequestDTO);
+                    logger.info(body);
+                })
+                .to("direct:lipana-transaction-status");
+
+
+        /*
+           Use this endpoint for receiving the callback form safaricom mpesa endpoint
+         */
+        from("rest:POST:/buygoods/callback")
+                .id("buy-goods-callback")
+                .setBody(exchange -> {
+                    String body = exchange.getIn().getBody(String.class);
+                    JSONObject object = new JSONObject(body);
+                    logger.info(body);
+                    // TODO IMPLEMENTATION
```

Comment:
@SubhamPramanik need help!!

Reply:
I would need some specifics. 

The implementation will be similar to: https://github.com/openMF/ph-ee-connector-gsma-mm/blob/fef9c82a11642cdf1fba762a5d02d909740ea940/src/main/java/org/mifos/connector/gsma/transfer/TransferRoutes.java#L110

https://github.com/openMF/ph-ee-connector-gsma-mm/blob/fef9c82a11642cdf1fba762a5d02d909740ea940/src/main/java/org/mifos/connector/gsma/transfer/TransferResponseProcessor.java#L35